handlers/cassandra/cassandra.go

```diff
@@ -116,6 +116,37 @@ func (h *Handler) GetE(cmd common.GetRequest) (<-chan common.GetEResponse, <-cha
 	dataOut := make(chan common.GetEResponse, len(cmd.Keys))
 	errorOut := make(chan error)
 
+	for idx, key := range cmd.Keys {
+		key_qi := func(q *gocql.QueryInfo) ([]interface{}, error) {
+			values := make([]interface{}, 1)
+			values[0] = key
+			return values, nil
+		}
+
+		var val []byte
+		var ttl uint32
+
+		if err := h.session.Bind("SELECT keycol,valuecol,TTL(valuecol) FROM kvstore.bucket1 where keycol=?", key_qi).Scan(&key, &val, &ttl); err == nil {
```

Comment:
You can batch those requests (unloged batch) instead of doing it one by one.
https://gist.github.com/0x6e6562/8728333
Just put a threshold to not batch too many requests at the same time